framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -341,4 +347,71 @@ public boolean validateType(Tree tree, AnnotatedTypeMirror type) {
 
         return true;
     }
+
+    /**
+     * Returns true if an expression of the given type can be a compile-time constant value.
+     *
+     * @param tm a type
+     * @return true if an expression of the given type can be a compile-time constant value
+     */
+    private boolean canBeConstant(TypeMirror tm) {
+        return TypesUtils.isPrimitive(tm)
+                || TypesUtils.isBoxedPrimitive(tm)
+                || TypesUtils.isString(tm)
+                || (tm.getKind() == TypeKind.ARRAY
+                        && canBeConstant(((ArrayType) tm).getComponentType()));
+    }
+
+    @Override
+    public Void visitMethod(MethodTree node, Void p) {
+        super.visitMethod(node, p);
+
+        ExecutableElement method = TreeUtils.elementFromDeclaration(node);
+        if (atypeFactory.getDeclAnnotation(method, StaticallyExecutable.class) != null) {
+            // The method is annotated as @StaticallyExecutable.
+            if (atypeFactory.getDeclAnnotation(method, Pure.class) == null) {
+                checker.reportWarning(node, "statically.executable.not.pure");
+            }
+            TypeMirror returnType = method.getReturnType();
+            if (returnType.getKind() != TypeKind.VOID && !canBeConstant(returnType)) {
+                checker.reportError(
+                        node, "statically.executable.nonconstant.return.type", returnType);
+            }
+
+            // Ways to determin the receiver type.
+            // 1. This definition of receiverType is null when receiver is implicit and method has
+            //    class com.sun.tools.javac.code.Symbol$MethodSymbol.  WHY?
+            //        TypeMirror receiverType = method.getReceiverType();
+            // 2. Can't use the tree, because the receiver might not be explicit.
+            // 3. Check whether method is static and use the declaring class.  Doesn't handle all
+            //    cases, but handles the most common ones.
+            TypeMirror receiverType = method.getReceiverType();
```

Comment:
The following should work:        
```
TypeMirror receiverType = TreeUtils.elementFromDeclaration(node).getReceiverType();
```

If the method has no receiver, the type returned with have `TypeKind.NONE`.

Reply:
That doesn't seem to work for me.  It returns a null value.  Please see https://dev.azure.com/mernst22/checker-framework/_build/results?buildId=5190&view=results .

Reply:
That's strange.  The documentation of ExecutableType does not match its implementation.  You can still use that method, just check for `null`.

Reply:
Or you can merge this pull request as is.